Improve empty state handling in query cubits (#56)
leancodepl/leancode_cubit_utils: packages/leancode_cubit_utils_cqrs/CHANGELOG.md

## 0.5.0

* **BREAKING CHANGES:** Remove `useQueryWithEmptyCubit` and `SimpleQueryWithEmptyCubit`.
* **BREAKING CHANGES:** Remove `useArgsQueryWithEmptyCubit` and `SimpleArgsQueryWithEmptyCubit`.

> * **BREAKING CHANGES:** Start using `isEmpty` checker of `useQueryCubit` and pass it `SimpleQueryCubit`.

Reviewer: This isn't a breaking change `isEmpty` was there so it's just a fix

> * **BREAKING CHANGES:** Add `isEmpty` checker to `useArgsQueryCubit` and `SimpleArgsQueryCubit`.

Reviewer: ```suggestion
* **BREAKING CHANGE:** Remove `useQueryWithEmptyCubit` and `SimpleQueryWithEmptyCubit`.
* **BREAKING CHANGE:** Remove `useArgsQueryWithEmptyCubit` and `SimpleArgsQueryWithEmptyCubit`.
* **BREAKING CHANGE:** Start using `isEmpty` checker of `useQueryCubit` and pass it `SimpleQueryCubit`.
* **BREAKING CHANGE:** Add `isEmpty` checker to `useArgsQueryCubit` and `SimpleArgsQueryCubit`.
```

> * **BREAKING CHANGES:** Add `isEmpty` checker to `useArgsQueryCubit` and `SimpleArgsQueryCubit`.

Reviewer: It's also not a breaking change. It's not required, so it's fully backwards compatible

## 0.5.0

> * Remove:

Reviewer: Mark those as breaking changes

> * Remove:

Reviewer: I'm not a huge fan of this nested formatting for changelogs

    - `useQueryWithEmptyCubit` and `SimpleQueryWithEmptyCubit`.
    - `useArgsQueryWithEmptyCubit`and `ArgsQueryWithEmptyCubit`.
* Update:

>     - Add `onEmpty` handler to `useQueryCubit` and `SimpleQueryCubit`

Reviewer: Technically, it wasn't added since it was already there, it  was just fixed

>     - Add `onEmpty` handler to `useArgsQueryWithEmptyCubit`and `ArgsQueryWithEmptyCubit`.

Reviewer: ```suggestion
    - Add `onEmpty` handler to `useArgsQueryCubit`and `ArgsQueryCubit`.
```

>     - Add `onEmpty` handler to `useArgsQueryWithEmptyCubit`and `ArgsQueryWithEmptyCubit`.

Reviewer: ```suggestion
    - Add `isEmpty` handler to `useArgsQueryWithEmptyCubit`and `ArgsQueryWithEmptyCubit`.
```
